### DISCUSSION TITLE
Use server SMTP for store e-mail

### DISCUSSION THREAD

**Discussion by iBobik** [btcpayserver/btcpayserver#5621] [Ideas / Feature Requests]
Currently each store requires to configure own SMTP settings, but it is very impractical if I am ok with using same settings for more stores - password cannot be recovered from existing store to copy it into new store and many SMTP servers cannot recover password too (just generate new password).

Could it be optional to configure SMTP? When not configured, it will use server global SMTP.

**Comment by pavlenex** [comment]
What do you mean by this ,I don't understand:

" password cannot be recovered from existing store to copy it into new store and many SMTP servers cannot recover password too (just generate new password)."

**Reply by iBobik** [reply]
Previously I was copying SMTP config from some existing store (because I want to use the same SMTP in multiple stores). Copy&paste server, port, login, sender and the password. But now the password cannot be accesses in already saved config, so I cannot configure it easily anymore.

**Reply by pavlenex** [reply]
Alright, I think the best approach here would be to have an option in Server Settings> Emails to use Server SMPT for all stores. Bear in mind, this can be abused if server is open for registration to everyone and malitious user registers and starts sending emails, that's the reason why it's store based right now.

**Reply by iBobik** [reply]
Aha, good to know. But I would allow to override it in the store settings, so by default stores use global SMTP, but can use their own.

**Reply by dennisreimann** [reply]
@pavlenex There's a server setting for that already in the Policies section. If available, stores will use the server SMTP as a fallback and the store's email section will mention that as of v1.12.

## Email Rules

![grafik](https://github.com/btcpayserver/btcpayserver/assets/886/5d911f8e-29f6-413e-be12-9deecd552582)

## Server Policies

![grafik](https://github.com/btcpayserver/btcpayserver/assets/886/0e43f2ee-316c-41fe-8ac2-d25aebf3b6e5)

**Reply by pavlenex** [reply]
Wow that's ... not clear UX, even I don't understand what this means.

**Reply by pavlenex** [reply]
Also the placing for this functionality is weird, are you really sure this is what it is? Who implemented this?

**Reply by iBobik** [reply]
I see different message in the store without SMTP configured:
![Snímek obrazovky 2023-12-27 v 0 35 10](https://github.com/btcpayserver/btcpayserver/assets/614232/e5e2176e-1cd8-456d-8166-0eb2c3b2e1e5)

**Reply by iBobik** [reply]
I would add message about the inheriting server's credentials here:
![Snímek obrazovky 2023-12-27 v 0 32 55](https://github.com/btcpayserver/btcpayserver/assets/614232/f3d1e8b1-98d5-4c32-ab85-9b1fe2cfdc20)
